### PR TITLE
Pending response bug dev push

### PR DESCRIPTION
**Pull request by nitish-beehyv**
Pending task closed fix feature pucardotorg#2032 (#1000)

Pending response bug fixed (#1009) 

## Release Notes

- **New Features**
  - Enhanced document upload functionality to support JPEG file types.
  - New asynchronous functions for fetching application data and handling submission responses in the Submission Review and View All Submissions components.
  
- **Improvements**
  - Updated event handling logic to accommodate different user roles and application statuses.

- **Configuration Changes**
  - Renamed action key from `RESPOND_TO_PRODUCTION_DOCUMENTS` to `PENDINGRESPONSE` in HomeConfig.

### CODE REVIEW

**Review by anirudh-0 — APPROVED** [review]

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/ViewAllSubmissions.js [hunk 1]
```diff
@@ -55,10 +66,14 @@
...
-            onClick={() => {
+            onClick={async () => {
               setShow(false);
               if (userRoles.includes("CITIZEN")) {
-                handleMakeSubmission(application);
+                if (application.status === "PENDINGRESPONSE") {
+                  const applicationData = await getApplication(application?.referenceId);
+                  setShow(false);
+                  openEvidenceModal(applicationData);
+                } else handleMakeSubmission(application);
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/ViewAllSubmissions.js:76]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/SubmissionsReview.js [hunk 2]
```diff
@@ -182,6 +183,20 @@
...
+      return response?.applicationList?.[0];
+    } catch (error) {
+      console.error("error :>> ", error);
+    }
+  };
+
+  const openResponseModal = async (applicationNumber) => {
+    const applicationData = await getApplication(applicationNumber);
+    docSetFunc(applicationData);
+  };
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/SubmissionsReview.js:198]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/SubmissionsReview.js [hunk 3]
```diff
@@ -182,6 +183,20 @@
...
     : applicationRes?.applicationList?.filter((application) => application.status === "PENDINGREVIEW");
 
+  const getApplication = async (applicationNumber) => {
+    try {
+      const response = await DRISTIService.searchSubmissions({ criteria: { filingNumber, applicationNumber, tenantId }, tenantId }, {}, "", true);
+      return response?.applicationList?.[0];
+    } catch (error) {
+      console.error("error :>> ", error);
+    }
+  };
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/SubmissionsReview.js:193]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/SubmissionsReview.js [hunk 4]
```diff
@@ -218,13 +233,15 @@
...
-                        app.status === "CREATE_SUBMISSION" ? "orderNumber" : "applicationNumber"
-                      }=${app.referenceId.split("_").pop()}`
-                    )
+                  ? app.status === "PENDINGRESPONSE"
+                    ? await openResponseModal(app.referenceId)
+                    : history.push(
+                        `/digit-ui/citizen/submissions/submissions-create?filingNumber=${filingNumber}&${
+                          app.status === "CREATE_SUBMISSION" ? "orderNumber" : "applicationNumber"
+                        }=${app.referenceId.split("_").pop()}`
+                      )
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/SubmissionsReview.js:244]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/citizen/FileCase/Config/resgisterRespondentConfig.js [hunk 5]
```diff
@@ -587,7 +587,7 @@ export const uploadResponseDocumentConfig = [
               name: "ID_Proof",
               validation: {},
               clearFields: { aadharNumber: "" },
-              allowedFileTypes: /(.*?)(png|jpg|pdf)$/i,
+              allowedFileTypes: /(.*?)(png|jpg|jpeg|pdf)$/i,
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/citizen/FileCase/Config/resgisterRespondentConfig.js:590]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/SubmissionsReview.js [hunk 6]
```diff
@@ -1,11 +1,12 @@
 import { Card } from "@egovernments/digit-ui-react-components";
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import { useTranslation } from "react-i18next";
 import useGetSubmissions from "../../../hooks/dristi/useGetSubmissions";
 import { CustomArrowOut } from "../../../icons/svgIndex";
 import EvidenceModal from "./EvidenceModal";
 import { useGetPendingTask } from "../../../../../home/src/hooks/useGetPendingTask";
 import { useHistory } from "react-router-dom";
+import { DRISTIService } from "../../../services";
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/SubmissionsReview.js:9]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/ViewAllSubmissions.js [hunk 7]
```diff
@@ -2,10 +2,12 @@ import React from "react";
 import Modal from "../../../components/Modal";
 import { CloseSvg } from "@egovernments/digit-ui-react-components";
 import { useHistory } from "react-router-dom";
+import { DRISTIService } from "../../../services";
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/ViewAllSubmissions.js:5]: checked

[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/ViewAllSubmissions.js:5]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/ViewAllSubmissions.js [hunk 8]
```diff
@@ -23,6 +25,15 @@
...
   };
 
+  const getApplication = async (applicationNumber) => {
+    try {
+      const response = await DRISTIService.searchSubmissions({ criteria: { filingNumber, applicationNumber, tenantId }, tenantId }, {}, "", true);
+      return response?.applicationList?.[0];
+    } catch (error) {
+      console.error("error :>> ", error);
+    }
+  };
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/ViewAllSubmissions.js:35]: checked

**Pull request merged**